### PR TITLE
Added OHLC support, applied golint.

### PR DESCRIPTION
**Pull request by sniggel**
This PR is a copy of https://github.com/beldur/kraken-go-api-client/pull/15 but rebase onto master (or master merged into it).

Features:
- Add OHLC support

Fixes:
- Applied `go lint`
- Made WithClient more consistent with a builder pattern

### CODE REVIEW

**Comment by sniggel** [comment]
Fingers crossed this PR is merged today :)

**Comment by sniggel** [comment]
bump

**Comment by sniggel** [comment]
bump

File: krakenapi.go [hunk 1]
```diff
@@ -91,24 +91,31 @@ const (
 	MinimumUSDT = 5.0
 )
 
-// KrakenApi represents a Kraken API Client connection
-type KrakenApi struct {
```
Comment (beldur) [inline krakenapi.go]: I understand that it should be `KrakenAPI` but wouldn't this be a breaking change?

File: krakenapi.go [hunk 2]
```diff
@@ -91,24 +91,31 @@ const (
...
+func New(key, secret string) *KrakenAPI {
+	krakenAPI := KrakenAPI{
+		key:    key,
+		secret: secret,
+		client: http.DefaultClient,
+	}
+	return &krakenAPI
 }
 
-func NewWithClient(key, secret string, httpClient *http.Client) *KrakenApi {
```
Comment (beldur) [inline krakenapi.go]: Same reasoning here. This looks like a breaking change to me.

File: krakenapi.go (hunk 1 above)
Comment (montpetitg) [reply krakenapi.go]: fixed

File: krakenapi.go (hunk 2 above)
Comment (montpetitg) [reply krakenapi.go]: fixed

File: krakenapi.go (hunk 1 above)
Comment (sniggel) [reply krakenapi.go]: fixed

File: krakenapi.go (hunk 2 above)
Comment (sniggel) [reply krakenapi.go]: fixed

**Comment by beldur** [comment]
Thanks for your contribution!

**Pull request merged**